CoreScriptsRoot/CoreScripts/Topbar.lua

```diff
@@ -1277,23 +1289,41 @@ local function Create3DUserGuiToggleIcon(topBarInstance, panel, menubar)
 		Image = "rbxasset://textures/ui/VR/toggle2D.png"
 	}
 
-	local userGuiPanel = Panel3D.Get("UserGui")
+	local thisModuleName = "UserGui"
+
+	local userGuiPanel = Panel3D.Get(thisModuleName)
 	userGuiPanel:SetType(Panel3D.Type.Fixed)
-	userGuiPanel:ResizePixels(300, 125)
+	userGuiPanel:ResizeStuds(4, 4, 128)
 	userGuiPanel:SetVisible(false, false)
 
-	menuItem.MouseButton1Click:connect(function()
-		if InputService.VREnabled then
-			userGuiPanel:SetVisible(not userGuiPanel.isVisible, false)
+	local VRHub = require(GuiRoot.Modules.VR.VRHub)
+
+	VRHub.ModuleOpened.Event:connect(function(moduleName, isExclusive, shouldCloseNonExclusive, shouldKeepTopbarOpen)
+		if shouldCloseNonExclusive and menuItem:IsActive() then
+			menuItem:SetActive(false)
+			userGuiPanel:SetVisible(false)
+
+			VRHub:FireModuleClosed(thisModuleName)
 		end
 	end)
 
+	function menuItem:OnClicked(wasActive)
+		menuItem:SetActive(not wasActive)
+		userGuiPanel:SetVisible(not wasActive)
+	
+		if not wasActive then
+			VRHub:FireModuleOpened(thisModuleName, false, false, false)
+		else
+			VRHub:FireModuleClosed(thisModuleName)
+		end
+	end
+
 	function userGuiPanel:OnVisibilityChanged(visible)
 		if visible then
 			local headLook = Panel3D.GetHeadLookXZ(true)
 			userGuiPanel.localCF = headLook * CFrame.Angles(math.rad(5), 0, 0) * CFrame.new(0, 0, 5)
 		end
-		menuItem:SetActive(visible)
+		menuItem:SetActive(visible, true)
```

Comment:
Is there a `SetActive` that takes 2 args? It looks like the initial definition only takes 1 arg?